src/components/EntitlementsPanelForm/Fields/Fields.tsx

```diff
@@ -0,0 +1,122 @@
+import { Select } from "@canonical/react-components";
+import { useQueryClient } from "@tanstack/react-query";
+import { useFormikContext } from "formik";
+
+import type { EntitlementSchema, EntityEntitlement } from "api/api.schemas";
+import { useGetResources } from "api/resources/resources";
+import CleanFormikField from "components/CleanFormikField";
+import FormikSubmitButton from "components/FormikSubmitButton";
+import { Endpoint } from "types/api";
+
+import { EntitlementsPanelFormLabel } from "..";
+
+import { Label } from "./types";
+
+type Props = {
+  entitlements: EntitlementSchema[];
+};
+
+const Fields = ({ entitlements }: Props) => {
+  const queryClient = useQueryClient();
+  const { values, setFieldValue, setFieldTouched } =
+    useFormikContext<EntityEntitlement>();
+  const { data: getResourcesData, isFetching: isGetResourcesFetching } =
+    useGetResources(
+      { entityType: values.entity_type },
+      { query: { enabled: !!values.entity_type } },
+    );
+  const resources = getResourcesData?.data.data || [];
+
+  return (
+    <div className="panel-table-form__fields">
+      <CleanFormikField
+        component={Select}
+        label={EntitlementsPanelFormLabel.ENTITY}
+        name="entity_type"
+        options={[
+          {
+            disabled: true,
+            label: "Select a resource type",
+            value: "",
+          },
+        ].concat(
+          [
+            ...new Set(
+              entitlements.map((entitlement) => entitlement.entity_type),
+            ),
+          ].map((entityType) => ({
+            disabled: false,
+            label: entityType,
+            value: entityType,
+          })),
+        )}
+        onChange={(event) => {
+          void setFieldValue("entity_type", event.target.value);
+          void setFieldValue("entity_id", "");
+          void setFieldTouched("entity_id", false);
+          void setFieldValue("entitlement", "");
+          void setFieldTouched("entitlement", false);
+          void queryClient.invalidateQueries({
+            queryKey: [Endpoint.RESOURCES],
+          });
+        }}
+      />
+      <CleanFormikField
+        component={Select}
+        label={EntitlementsPanelFormLabel.RESOURCE}
+        name="entity_id"
+        disabled={!values.entity_type}
+        options={[
+          {
+            disabled: true,
+            label: isGetResourcesFetching
+              ? Label.LOADING_RESOURCES
+              : Label.SELECT_RESOURCE,
+            value: "",
+          } as any,
+        ].concat(
+          resources.map((resource) => ({
+            disabled: false,
+            label: `${resource.entity.name} (${resource.entity.id})`,
```

Comment:
Did you end up checking with Lukas about this? Any idea how long these names and IDs could be?